app/HutchManager/Services/ResultsModifierService.cs

```diff
@@ -20,6 +20,16 @@ public async Task<List<Models.ResultsModifierModel>> List()
       .Include(x => x.Type)
       .Include(x => x.ActivitySource)
       .ToListAsync();
+    foreach (ResultsModifier r in list)
+    {
+      r.ActivitySource = await _db.ActivitySources
+                           .AsNoTracking()
+                           .Include(x => x.Type)
+                           .Include(x => x.TargetDataSource)
+                           .SingleOrDefaultAsync(x => x.Id == r.ActivitySource.Id)
+                         ?? throw new KeyNotFoundException();
+    }
+
```

Comment:
use `ThenInclude()` to include relations of already included relations.
Doing it on the same IQueryable will do it in a single database call, instead of `n + 1` calls where `n` is the number of results modifiers.

```suggestion
      .Include(x => x.ActivitySource)
        .ThenInclude(x => x.Type)
      .Include(x => x.ActivitySource)
        .ThenInclude(x => x.TargetDataSource)
      .ToListAsync();
```

confirm it works though; my syntax above might not be perfect; don't use `ThenInclude` that often 😅